x-pack/legacy/plugins/reporting/export_types/printable_pdf/server/lib/generate_pdf.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as Rx from 'rxjs';
+import { toArray, mergeMap } from 'rxjs/operators';
+import moment from 'moment-timezone';
+import { groupBy } from 'lodash';
+import { KbnServer } from '../../../../types';
+// @ts-ignore
+import { pdf } from './pdf';
+// @ts-ignore
+import { oncePerServer } from '../../../../server/lib/once_per_server';
+import { screenshotsObservableFactory } from '../../../common/lib/screenshots';
+import { createLayout } from '../../../common/layouts';
+import { TimeRange } from '../../../common/lib/screenshots/types';
+import { LayoutInstance } from '../../../common/layouts/layout';
+
+interface ScreenshotData {
+  base64EncodedData: string;
+  title: string;
+  description: string;
+}
+
+interface UrlScreenshot {
+  screenshots: ScreenshotData[];
+  timeRange: TimeRange;
+}
+
+const getTimeRange = (urlScreenshots: UrlScreenshot[]) => {
+  const grouped = groupBy(urlScreenshots.map(u => u.timeRange));
+  const values = Object.values(grouped);
+  if (values.length === 1) {
+    return values[0][0];
+  }
+
+  return null;
+};
+
+const formatDate = (date: Date, timezone: string) => {
+  return moment.tz(date, timezone).format('llll');
+};
+
+function generatePdfObservableFn(server: KbnServer) {
+  const screenshotsObservable = screenshotsObservableFactory(server);
+  const captureConcurrency = 1;
+
+  return function generatePdfObservable(
+    title: string,
+    urls: string[],
+    browserTimezone: string,
+    conditionalHeaders: any,
```

Comment:
I think we can replace any in here? maybe a `Record<string, string>`, or `string[]`

Reply:
Oh snap, we already had a type def in the code for `ConditionalHeaders` :) thanks Ahmad 

85b68acb136453fe4fdf08ac23407663911d40c3